src/utils/fetch.ts

```diff
@@ -0,0 +1,117 @@
+/**
+ * This module exports an instance of fetch augmented with
+ * timeout and retries with exponential backoff.
+ */
+
+import { left, right } from 'fp-ts/lib/Either';
+import { fromEither, TaskEither } from 'fp-ts/lib/TaskEither';
+import { calculateExponentialBackoffInterval } from 'italia-ts-commons/lib/backoff';
+import { AbortableFetch, retriableFetch, setFetchTimeout, toFetch } from 'italia-ts-commons/lib/fetch';
+import { RetriableTask, TransientError, withRetries } from 'italia-ts-commons/lib/tasks';
+
+import { Millisecond } from 'italia-ts-commons/lib/units';
+import nodeFetch from 'node-fetch';
+
+// The old following import has been substituted by an explicit declaration
+// { fetchMaxRetries, fetchTimeout } from "../config";
+
+const fetchMaxRetries = 5;
+const fetchTimeout: Millisecond = 1000 as Millisecond;
+
+//
+// Returns a fetch wrapped with timeout and retry logic
+//
+
+function retryingFetch(fetchApi: typeof fetch, timeout: Millisecond, maxRetries: number): typeof fetch {
+  // a fetch that can be aborted and that gets cancelled after fetchTimeoutMs
+  const abortableFetch = AbortableFetch(fetchApi);
+  const timeoutFetch = toFetch(setFetchTimeout(timeout, abortableFetch));
+  // configure retry logic with default exponential backoff
+  // @see https://github.com/pagopa/io-ts-commons/blob/master/src/backoff.ts
+  const exponentialBackoff = calculateExponentialBackoffInterval();
+  const retryLogic = withRetries<Error, Response>(maxRetries, exponentialBackoff);
+  const retryWithTransient429s = retryLogicForTransientResponseError((_: any) => _.status === 429, retryLogic);
+  return retriableFetch(retryWithTransient429s)(timeoutFetch);
+}
+
+//
+// Default fetch configured with a short timeout and an exponential backoff
+// retrying strategy - suitable for calling the backend APIs that are supposed
+// to respond quickly.
+
+export function defaultRetryingFetch(timeout: Millisecond = fetchTimeout, maxRetries: number = fetchMaxRetries) {
+  // Override default react-native fetch with whatwg's that supports aborting
+
+  // NOTE: In fact, react-native supports Aborting Controller. Please check
```

Comment:
What's the point of having this comment here? Perhaps we should open a story in the app backlog

Reply:
The comment explains the goal of the function `defaultRetryingFetch` in the IO App. It's an old comment, I should have removed it, as `iopay` stack is not based on `react-native`. We already have a story to add the `Abort Controller` to fetches https://www.pivotaltracker.com/story/show/176339507